File: longest-subarray-of-1s-after-deleting-one-element/longest-subarray-of-1s-after-deleting-one-element.ts
function longestSubarray(nums: number[]): number {
    let prev = 0, cur = 0, max = 0;
    for(const n of nums) {
        if(n === 0) {
            max = Math.max(prev + cur, max);
            prev = cur;
            cur = 0;
        } else {
            cur++;
        }
    }
    if(cur === nums.length) return cur - 1
    return Math.max(max, prev + cur);
};

File: 1586-longest-subarray-of-1s-after-deleting-one-element/longest-subarray-of-1s-after-deleting-one-element.rs
impl Solution {
    pub fn longest_subarray(nums: Vec<i32>) -> i32 {
        let mut a = 0;
        let mut b = 0;
        let mut max = 0;
        for n in nums.iter() {
            if *n == 1 {
                b += 1;
            } else {
                max = max.max(a + b);
                a = b;
                b = 0;
            }
        }
        if b == nums.len() as i32 { b - 1 } else { max.max(a + b) }
    }
}